rflx/pyrflx/typevalue.py

```diff
@@ -61,25 +81,19 @@ def literals(self) -> Mapping[Name, Expr]:
         raise NotImplementedError
 
     @classmethod
-    def construct(cls, vtype: Type) -> "TypeValue":
+    def construct(cls, vtype: Type, all_messages: List[model.Message]) -> "TypeValue":
         if isinstance(vtype, Integer):
             return IntegerValue(vtype)
         if isinstance(vtype, Enumeration):
             return EnumValue(vtype)
         if isinstance(vtype, Opaque):
             return OpaqueValue(vtype)
+        if isinstance(vtype, Array):
+            return ArrayValue(vtype, all_messages)
+        if isinstance(vtype, model.Message):
+            return MessageValue(vtype, all_messages)
         raise ValueError("cannot construct unknown type: " + type(vtype).__name__)
```

Comment:
I added the default value but I don't think it would be practical to throw an Argument Error. The reason being that in the constructor of MessageValue all Fields of the message are created at once and  TypeValue.construct is used to create the instances of TypeValue used to instantiate the Fields. The all_messages parameter is passed each time a Field (and therefore a TypeValue) is created and holds a list of all known model.messages in case the TypeValue is an ArrayValue.

Reply:
Fixed in https://github.com/Componolit/RecordFlux/pull/198/commits/1ab7b6311e8bc2f82f25c3c4b96d70a135337223.